### PR TITLE
Display toast notification on editor distance spacing changes

### PR DESCRIPTION
**Pull request by frenzibyte**
- [x] Depends on #18070

Fulfills the second part of https://github.com/ppy/osu/pull/16576#issuecomment-1115067772, for better visibility of the distance spacing value to the mapper (also no longer expands the setting when pressing the hotkey, given that it's main purpose is to show that the spacing is going to be changed).

PR'd separately as I'm not 100% sure on the direction taken UI-wise, maybe good enough for now?

### CODE REVIEW

File: osu.Game/Rulesets/Edit/DistancedHitObjectComposer.cs [hunk 1]
```diff
@@ -168,5 +181,26 @@ public ExpandingToolboxContainer()
...
+            public DistanceSpacingToast(LocalisableString value, ValueChangedEvent<double> change)
+                : base(getAction(change).GetLocalisableDescription(), value, string.Empty)
+            {
+                this.change = change;
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OsuConfigManager config)
+            {
+                ShortcutText.Text = config.LookupKeyBindings(getAction(change));
```
Comment (frenzibyte) [inline osu.Game/Rulesets/Edit/DistancedHitObjectComposer.cs:198]: This is pretty bad, but similar to what we used in `MusicKeyBindingHandler` as well. Open to moving that to a better method, not sure where it should be placed though. `OsuGame`? or maybe that method shouldn't exist in the first place and let each usage check `RealmKeyBindingStore.GetReadableCombination`.

Comment (bdach) [reply osu.Game/Rulesets/Edit/DistancedHitObjectComposer.cs:198]: Not sure what's so bad about this? I don't see the issue.

I do see an issue however in that `MusicKeyBindingHandler` generally displays the binding uppercased:

```suggestion
                ShortcutText.Text = config.LookupKeyBindings(getAction(change)).ToUpper();
```

Comment (frenzibyte) [reply osu.Game/Rulesets/Edit/DistancedHitObjectComposer.cs:198]: Bad is that it resolves `OsuConfigManager` to look up for key binding strings, the operation is completely irrelevant from the class.

Comment (bdach) [reply osu.Game/Rulesets/Edit/DistancedHitObjectComposer.cs:198]: I'd rather have this kind of glue code duplicated than call some abstract monstrosity into being, quite frankly. It's like three lines.

**Review by bdach — DISMISSED** [review]

File: osu.Game/Rulesets/Edit/DistancedHitObjectComposer.cs [hunk 2]
```diff
@@ -168,5 +181,26 @@ public ExpandingToolboxContainer()
...
+
+            [BackgroundDependencyLoader]
+            private void load(OsuConfigManager config)
+            {
+                ShortcutText.Text = config.LookupKeyBindings(getAction(change)).ToUpper();
+            }
+
+            private static GlobalAction getAction(ValueChangedEvent<double> change) => change.NewValue - change.OldValue > 0
+                ? GlobalAction.EditorIncreaseDistanceSpacing
+                : GlobalAction.EditorDecreaseDistanceSpacing;
```
Comment (peppy) [inline osu.Game/Rulesets/Edit/DistancedHitObjectComposer.cs:203]: I guess this works okay. Feels... interesting that it shows the "decrease" / "increase" when adjusting using the slider, but I don't necessarily think it's a bad thing as it exposes the two shortcuts.

**Review by peppy — DISMISSED** [review]

**Review by bdach — APPROVED** [review]

File: osu.Game/Rulesets/Edit/DistancedHitObjectComposer.cs (hunk 2 above)
Comment (bdach) [reply osu.Game/Rulesets/Edit/DistancedHitObjectComposer.cs:203]: UI scale slider does this, too, and so does the mania scroll speed one. So it's consistent at least.

**Pull request merged**